keps/sig-network/2086-service-internal-traffic-policy/README.md

```diff
@@ -123,51 +119,46 @@ type ServiceSpec struct {
-* if `externalTrafficPolicy=Local`, `externalTrafficPolicy` will take precedent over `trafficPolicy`, but only for external sources.
+This field will be independent from externalTrafficPolicy. In other words, internalTrafficPolicy only applies to traffic originating from internal sources.
 
 Proposed changes to kube-proxy:
-* when `trafficPolicy=Cluster`, default to existing behavior today.
-* when `trafficPolicy=Topology`, use topology hints from EndpointSlice API.
-* when `trafficPolicy=PreferLocal`, route to endpoints in EndpointSlice that matches the local node's topology (topology defined by `kubernetes.io/hostname`),
-fall back to "Cluster" behavior if there are no local endpoints.
-* when `trafficPolicy=Local`, route to endpoints in EndpointSlice that maches the local node's topology, drop traffic if none exist.
+* when `internalTrafficPolicy=Cluster`, default to existing behavior today.
+* when `internalTrafficPolicy=Local`, route to endpoints in EndpointSlice that maches the local node's topology, drop traffic if none exist.
 
 ### Test Plan
 
 Unit tests:
-* unit tests validating API strategy/validation for when `trafficPolicy` is set on Service.
-* unit tests exercising kube-proxy behavior when `trafficPolicy` is set to all possible values.
+* unit tests validating API strategy/validation for when `internalTrafficPolicy` is set on Service.
+* unit tests exercising kube-proxy behavior when `internalTrafficPolicy` is set to all possible values.
 
 E2E test:
-* e2e tests validating default behavior with kube-proxy did not change when `trafficPolicy` defaults to `Cluster`. Existing tests should cover this.
-* e2e tests validating that traffic is preferred to local endpoints when `trafficPolicy` is set to `PreferLocal`.
-* e2e tests validating that traffic is only sent to node-local endpoints when `trafficPolicy` is set to `Local`.
+* e2e tests validating default behavior with kube-proxy did not change when `internalTrafficPolicy` defaults to `Cluster`. Existing tests should cover this.
+* e2e tests validating that traffic is only sent to node-local endpoints when `internalTrafficPolicy` is set to `Local`.
 
 ### Graduation Criteria
 
 Alpha:
-* feature gate `ServiceTrafficPolicy` _must_ be enabled for apiserver to accept values for `spec.trafficPolicy`. Otherwise field is dropped.
-* kube-proxy handles traffic routing for 4 initial internal traffic policies `Cluster`, `Topology`, `PreferLocal` and `Local`.
+* feature gate `ServiceInternalTrafficPolicy` _must_ be enabled for apiserver to accept values for `spec.internalTrafficPolicy`. Otherwise field is dropped.
+* kube-proxy handles traffic routing for 2 initial internal traffic policies `Cluster`, and `Local`.
 * Unit tests as defined in "Test Plan" section above. E2E tests are nice to have but not required for Alpha.
 
+Beta:
+* integration tests exercising API behavior for `spec.internalTrafficPolicy` field of Service.
```

Comment:
I don't see these in the test plan above, do you have any specific ones you're thinking of?

Reply:
No, I haven't thought that deep into the details yet.

Reply:
I can't see any valuable integration test for this field ... should be better not committing to it?
we can always add them later but this will stop the graduation

Reply:
I kind of agree with Antonio

Reply:
Me too, will re-evalute whether integration tests are worth adding during PR